ggml-cuda.cu

```diff
@@ -512,6 +512,13 @@ static size_t g_scratch_offset = 0;
 
 static cublasHandle_t g_cublas_handles[GGML_CUDA_MAX_DEVICES] = {nullptr};
 
+static __device__ void bad_arch() {
+    printf("ERROR: ggml-cuda was compiled without support for the current GPU architecture.\n");
```

Comment:
```suggestion
    fputs("ERROR: ggml-cuda was compiled without support for the current GPU architecture.\n", stderr);
```

Reply:
It's pretty awesome that `printf` can be used in CUDA kernels, but I don't think that these are available. Only the basic `printf`.

Reply:
So assert is the only way to print to stderr. Printing to stdout is not very nice for CLI apps using llama.cpp, but I can't think of a good way to fix this.

Reply:
Keep in mind that this is printed once in every thread, and there are a lot of them. There is nothing to do other than rebuild with the right architectures if you encounter this error. It looks like this.
![image](https://github.com/ggerganov/llama.cpp/assets/2141330/cf606fdc-5dd7-4352-8bd6-0d5e22eb9053)